components/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/WidgetInfoProviderAPI.java

```diff
@@ -40,46 +41,57 @@
 
     /**
      * This method returns a list of meta information of widgets.
+     *
      * @return List of widget related meta information
      */
     @GET
-    @Path("/metainfo")
-    @Produces("application/json")
-    public List getWidgetsMetaInfo() {
+    @Path("/")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getWidgetsMetaInfo() {
         List<WidgetInfoProvider> widgetInfoProviderImplList = WidgetDataHolder.getInstance().getWidgetInfoProviders();
-        List list = widgetInfoProviderImplList.stream().map(widget -> widget.getWidgetsMetaInfo())
-                .filter(Objects::nonNull).collect(Collectors.toList());
-        return list;
+        List list = widgetInfoProviderImplList.stream()
+                .map(widget -> widget.getWidgetsMetaInfo())
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        return Response.ok().entity(list).build();
     }
 
     /**
      * This method return the configuration of given widget.
+     *
      * @param widgetId widget id
-     * @return  widget configuration
+     * @return widget configuration
      */
     @GET
-    @Path("/conf/{id}")
-    @Produces("application/json")
-    public String getWidgetConf(@PathParam("id") String widgetId) {
+    @Path("/{id}/conf")
```

Comment:
What is the reason for changing /conf/{id} to /{id}/conf? IMO /conf/{id} is more cleaning and easy to understand.